google_api_lib/utils.py

```diff
@@ -5,12 +5,44 @@
 from google.oauth2 import service_account
 
 from celery import Task
+import re
+import logging
+
+logger = logging.getLogger(__name__)
 
 
 def enabled():
     return settings.GOOGLE_API_AUTHN_INFO is not None
 
 
+def extract_id_from_sheets_url(url):
+    """
+    Assumes `url` is of the form
+    https://docs.google.com/spreadsheets/d/<ID>/...
+    and returns the <ID> portion
+    """
+    pattern = ".*docs\.google\.com\/spreadsheets\/d\/([-\w]+).*"
+    m = re.search(pattern, url)
+    if m is None:
+        logger.error(f"Invalid url: {url}")
+        return ""
+    return m.group(1)
+
+
+def extract_id_from_drive_folder_url(url):
+    """
+    Assumes `url` is of the form
+    https://drive.google.com/drive/u/2/folders/<ID>/..
+    and returns the <ID> portion
+    """
+    pattern = ".*drive\.google\.com\/drive\/u\/2\/folders\/([-\w]+).*"
```

Comment:
I believe these `/u/<n>/` URLs mean you are logged into multiple Google accounts -- /u/1 is the first account you're logged in to, /u/2 is the second, etc. This is going to be different per person depending on what accounts they're logged in to.

I think you want to use the canonical drive id, which should look something like https://drive.google.com/drive/folders/<id>...

(You can navigate to a folder in Drive and Get Link to get the canonical link)